gpii/node_modules/solutionsRegistry/src/js/filter-schema-settings.js

```diff
@@ -0,0 +1,35 @@
+/* eslint-env node */
+"use strict";
+var fluid = require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+
+fluid.registerNamespace("gpii.universal.solutionsRegistry");
+
+/**
+ *
+ * Filter a settings schema so that it only has checks ("properties") for the specific observed settings.  Used in
+ * the sanity checks of the solutions registry itself, and in settings validation middleware.
+ *
+ * @param {Object} observedSettings - The observed settings that we should use to filter the schema.
+ * @param {Object} settingsSchema - The schema to filter.
+ * @return {Object} The filtered schema.
+ *
+ */
+gpii.universal.solutionsRegistry.filterSchemaToSettings = function (observedSettings, settingsSchema) {
+    if (typeof observedSettings === "object") {
+        var filteredSchema     = fluid.copy(settingsSchema);
+        var requiredKeys       = fluid.keys(observedSettings);
+        var allProperties      = fluid.get(settingsSchema, "properties");
+        var filteredProperties = fluid.filterKeys(allProperties, requiredKeys);
+        if (Object.keys(filteredProperties).length) {
```

Comment:
We tend to keep around $.isEmptyObject for this purpose

Reply:
Once I figured out that I could use `fluid.jQueryStandalone.isEmptyObject` from Node, I updated  both usages of this pattern.